### ISSUE TITLE
Auto-label Issues or PR with a specified label

### ISSUE THREAD

**Issue by shine2lay** [mergeability/mergeable#59]
*If you want this feature, please react with a 👍*

---
We want to be able to label an issue or pull request when an event occurs. There are two scenarios where this applies:

1. When an event occurs under certain conditions (validation) -- example: Label based on the inclusion of certain keywords in the description of the issue / PR.
2. When an event occurs -- example: Auto-label when an issue is opened (i.e. with the label "triage")

Implementing a `Label` Action class will automatically make scenario `1` possible:
```yml
version: 2
mergeable:
  - when: issues.opened
    validate:
      - do: description
        must_include:
          regex: 'Depends on'
    pass:
      - do: label
        name: Dependencies
```

For scenario `2` there isn't a good way to write the recipe. Ideally if we don't want to spcify any validations we should be able to execute any actions whenever an event occurs:

```yml
when: issues.opened
   execute:
      - do: label
         name: 'triage'
```

Without the execute block we may be able to write it as:

```yml
when: issues.opened
   validate:
   pass:
      - do: label
         name: 'triage'
```

The above should work (I have not verified) but semantically clunky.... 

**Question**: Are there other use cases that fits our current workflow architecture? 

# History

*Below is the original description for this issue written based on the older version of mergeable*

Context
---
When new issues are created, auto-label them with a specified label i.e `triage` this will allow repo managers to easily identify which issues need to be triaged

Solution
---
``` yml
issues:
  opened:
    add-label:
      label: 'triage' // create label if not found
      only-when-empty: true | false // only add label if no other label exists
```

**Comment by jusx** [comment]
Here is a use case that can be useful to have.

It will be really useful to be able to label issues as "in progress" or "claimed" so that folks are aware that an issue is being worked on. When the GH keywords like `closes` is used in a PR. For example, if a PR is notated as "closes #1" then Issue number 1 should be auto-labeled with "in progress"

There are two ways to make it possible for Actions to do these kinds of things:
1. Parse descriptions of issues and pull requests for GH keywords and pass them into Mergeable `Actions`  so that actions can do things with it.
2. Pass the payload to the `Actions` so that it can do whatever it needs to do...

**Comment by jusx** [comment]
Resolved by #211
